rust/ql/test/library-tests/type-inference/pattern_matching.rs

```diff
@@ -478,7 +478,7 @@ pub fn tuple_patterns() {
     let single = (42i32,);
     match single {
         (x,) => {
-            let single_elem = x; // $ MISSING: type=single_elem:i32
+            let single_elem = x; // $ type=single_elem:i32
```

Comment:
Out of interest, what is the second element of `single`?

Reply:
There isn't a second element . The trailing `,` is to distinguish between a parenthesized expression and a singleton tuple. `(1)` has type `i32` but `(1,)` has type `(i32)`.

Reply:
I see, thanks.